Base/Logic/vtkSlicerApplicationLogic.cxx

```diff
@@ -1728,6 +1728,34 @@ bool vtkSlicerApplicationLogic::IsPluginInstalled(const std::string& filePath,
   return true;
 }
 
+//----------------------------------------------------------------------------
+bool vtkSlicerApplicationLogic::IsPluginBuiltIn(const std::string& filePath,
+                                                  const std::string& applicationHomeDir)
+{
+  if (filePath.empty())
+    {
+    vtkGenericWarningMacro( << "filePath is an empty string !");
+    return false;
+    }
+  if (applicationHomeDir.empty())
+    {
+    vtkGenericWarningMacro( << "applicationHomeDir is an empty string !");
+    return false;
+    }
+
+  std::string path = itksys::SystemTools::GetFilenamePath(filePath);
+  std::string canonicalPath = itksys::SystemTools::GetRealPath(path.c_str());
+
+  // On MacOSX extensions are installed in the "<Slicer_EXTENSIONS_DIRBASENAME>-<slicerRevision>"
+  // folder being a sub directory of the application dir, an extra test is required to make sure the
+  // tested filePath doesn't belong to that "<Slicer_EXTENSIONS_DIRBASENAME>-<slicerRevision>" folder.
+  // Since package name can be rename from "Slicer.app" to "Something.app", let's compare
+  // using ".app/Contents/" instead of "Slicer_BUNDLE_LOCATION" which is "Slicer.app/Contents/"
+  bool macExtension = (canonicalPath.find(".app/Contents/" Slicer_EXTENSIONS_DIRBASENAME "-") != std::string::npos);
+
+  return itksys::SystemTools::StringStartsWith(canonicalPath.c_str(), applicationHomeDir.c_str()) || macExtension;
```

Comment:
Should it be 
```
return itksys::SystemTools::StringStartsWith(canonicalPath.c_str(), applicationHomeDir.c_str()) && !macExtension;
```

To confirm, could you add a test similar to the one for `isPluginInstalled` ? 
See https://github.com/cpinter/Slicer/blob/70196a7d2c1414c2cc51a5ce805eea0642923134/Base/QTCore/Testing/Cxx/qSlicerUtilsTest1.cxx#L271-L273